code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -31,90 +31,76 @@
 
 /datum/reagent/proc/reaction_mob(var/mob/living/M, var/method = TOUCH, var/volume)
 
-	if(!holder)
-		return 1
-	if(!istype(M))
-		return 1
-
 	var/datum/reagent/self = src //Note : You need to declare self again (before the parent call) to use it in your chemical, see blood
 	src = null
 
+	if(!istype(M))
+		return 1
+
 	//If the chemicals are in a smoke cloud, do not let the chemicals "penetrate" into the mob's system (balance station 13) -- Doohl
-	if(self.holder && !istype(self.holder.my_atom, /obj/effect/effect/smoke/chem))
-		if(method == TOUCH)
+	if(method == TOUCH)
 
-			var/chance = 1
-			var/block  = 0
+		var/chance = 1
+		var/block  = 0
 
-			for(var/obj/item/clothing/C in M.get_equipped_items())
-				if(C.permeability_coefficient < chance)
-					chance = C.permeability_coefficient
+		for(var/obj/item/clothing/C in M.get_equipped_items())
+			if(C.permeability_coefficient < chance)
+				chance = C.permeability_coefficient
 
-				//Hardcode, but convenient until protection is fixed
-				if(istype(C, /obj/item/clothing/suit/bio_suit))
-					if(prob(75))
-						block = 1
+			//Hardcode, but convenient until protection is fixed
+			if(istype(C, /obj/item/clothing/suit/bio_suit))
+				if(prob(75))
+					block = 1
 
-				if(istype(C, /obj/item/clothing/head/bio_hood))
-					if(prob(75))
-						block = 1
+			if(istype(C, /obj/item/clothing/head/bio_hood))
+				if(prob(75))
+					block = 1
 
-			chance = chance * 100
+		chance *= 100
 
-			if(prob(chance) && !block)
-				if(M.reagents)
-					M.reagents.add_reagent(self.id, self.volume/2) //Hardcoded, transfer half of volume
+		if(prob(chance) && !block)
+			if(M.reagents)
+				M.reagents.add_reagent(self.id, self.volume/2) //Hardcoded, transfer half of volume
 
 /datum/reagent/proc/reaction_animal(var/mob/living/simple_animal/M, var/method=TOUCH, var/volume)
 
-	if(!holder)
-		return 1
-	if(!istype(M))
-		return 1
 
 	var/datum/reagent/self = src
 	src = null
 
+	if(!istype(M))
+		return 1
+
 	M.reagent_act(self.id, method, volume)
 
 /datum/reagent/proc/reaction_obj(var/obj/O, var/volume)
 
-	if(!holder)
-		return 1
+	src = null
```

Comment:
What the fuck is this
Why is it nulled AGAIN after an if() check on an unrelated thing